jablib/src/main/java/org/jabref/logic/remote/RemotePreferences.java

```diff
@@ -77,9 +95,19 @@ public static InetAddress getIpAddress() throws UnknownHostException {
         return InetAddress.getByName("localhost");
     }
 
+    private boolean isUriValid(String protocol, String hostAddress, int port) {
```

Comment:
Try to use this in a validator... 

Create a method accepting host AND port.  - Nothing else